src/CommunityToolkit.Maui/Essentials/AppTheme/AppThemeObjectOfT.shared.cs

```diff
@@ -0,0 +1,49 @@
+﻿namespace CommunityToolkit.Maui;
+
+/// <summary>
+/// Represents an object that is aware of the operating system theme.
+/// </summary>
+public abstract class AppThemeObject<T>
```

Comment:
Shall we rename this to `ThemeObject<T>`?

After renaming `ThemeResourceExtension`, it's a bit confusing to be using `<mct:AppThemeColor ...`, `<mct:AppThemeResource ...`, and `{mct:ThemeResource ...` in XAML. 

I assume devs would prefer consistency of each being named `Theme*`